File: srm204/medici.py
class Medici:
	def winner(self, fame, fortune, secrets): 
		scores = [0 for x in range(len(fame))]
		for i in range(len(scores)):
			scores[i] = min(fame[i], fortune[i], secrets[i])
		
		res = scores.index(max(scores))
		if scores.count(max(scores)) > 1: return -1
		return res

File: srm204/Medici.java
import java.lang.*;

public class Medici {

	public int winner(int[] fame, int[] fortune, int[] secrets) {
		int res = 0;
		int max = 0;
		int count = 0;
		int n = fame.length;
		int[] scores = new int[n];
		for (int i = 0; i < fame.length; i++) {
			int tmp = Math.min(fame[i], fortune[i]);
			scores[i] = Math.min(tmp, secrets[i]);
		}
		for (int j = 0; j < scores.length; j++) {
			if (max < scores[j]) {
				max = scores[j];
				res = j;
			}
		}
		for (int k = 0; k < scores.length; k++) {
			if (max == scores[k]) {
				count++;
			}
		}
		if (count > 1) {
			res = -1;
		}
	return res;
	}
}